frontend/apps/filemanager/filemanager.lua

```diff
@@ -522,35 +519,66 @@ function FileManager:pasteHere(file)
         local orig = util.realpath(self.clipboard)
         local dest = lfs.attributes(file, "mode") == "directory" and
             file or file:match("(.*/)")
-        if self.cutfile then
+
+        local function infoCopyFile()
+            -- if we copy a file, also copy its sidecar directory
+            if DocSettings:hasSidecarFile(orig) then
+                util.execute(self.cp_bin, "-r", DocSettings:getSidecarDir(orig), dest)
+            end
+            if util.execute(self.cp_bin, "-r", orig, dest) == 0 then
+                UIManager:show(InfoMessage:new {
+                    text = T(_("Copied to: %1"), dest),
+                    timeout = 2,
+                })
+            else
+                UIManager:show(InfoMessage:new {
+                    text = T(_("An error occurred while trying to copy %1"), orig),
+                    timeout = 2,
+                })
+            end
+        end
+
+        local function infoMoveFile()
             -- if we move a file, also move its sidecar directory
             if DocSettings:hasSidecarFile(orig) then
                 self:moveFile(DocSettings:getSidecarDir(orig), dest) -- dest is always a directory
             end
-            self:moveFile(orig, dest)
-        else
-            if DocSettings:hasSidecarFile(orig) then
-                util.execute(self.cp_bin, "-r", DocSettings:getSidecarDir(orig), dest)
+            if self:moveFile(orig, dest) then
+                UIManager:show(InfoMessage:new {
+                    text = T(_("Moved to: %1"), dest),
+                    timeout = 2,
+                })
+            else
+                UIManager:show(InfoMessage:new {
+                    text = T(_("An error occurred while trying to move %1"), orig),
+                    timeout = 2,
+                })
             end
             util.execute(self.cp_bin, "-r", orig, dest)
         end
-    end
-end
 
-function FileManager:createFolder(curr_folder, new_folder)
-    local folder = curr_folder .. "/" .. new_folder
-    local code = util.execute(self.mkdir_bin, folder)
-    local text
-    if code == 0 then
-        self:onRefresh()
-        text = T(_("Folder created:\n %1"), new_folder)
-    else
-        text = _("The folder has not been created")
+        local info_file
+        if self.cutfile then
+            info_file = infoMoveFile
+        else
+            info_file = infoCopyFile
+        end
+        local basename = util.basename(self.clipboard)
+        local mode = lfs.attributes(dest .."/" .. basename, "mode")
+        if mode == "file" or mode == "directory" then
+            UIManager:show(ConfirmBox:new {
+                text = T(_("The %1 %2 already exists. Do you want to overwrite it?"), mode, basename),
```

Comment:
I don't think this is the best idea for localization, setting aside the fact that neither file nor directory would be translated as currently written. Redundancy is almost always better in these matters. Even just in Dutch:

>The file blabla already exists.
*Het* bestand blabla bestaat al.

>The folder/directory already exists.
*De* map/directory bestaat al.

Similar in German with different genders. Die Datei (file) and der Ordner (folder).